dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -74,12 +76,50 @@ public CommunityRest findOne(Context context, UUID id) {
 		Page<CommunityRest> page = new PageImpl<Community>(communities, pageable, total).map(converter);
 		return page;
 	}
+
+	// TODO: Add methods in dspace api to support pagination of top level
+	// communities
+	@SearchRestMethod(name="top")
+	public Page<CommunityRest> findAllTop(Pageable pageable) {
+		List<Community> topCommunities = new ArrayList<Community>();
+		int total = 0;
+		try {
+			List<Community> it = cs.findAllTop(obtainContext());
+			total = it.size();
+			for (Community c : it) {
+				topCommunities.add(c);
+			}
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<CommunityRest> page = utils.getPage(topCommunities, pageable).map(converter);
+		return page;
+	}
+
+	// TODO: add method in dspace api to support direct query for subcommunities
+	// with pagination and authorization check
+	@SearchRestMethod(name="subCommunities")
```

Comment:
Why didn't we add subcommunities to the `org.dspace.app.rest.model.CommunityRest` and embedded them in the `org.dspace.app.rest.model.hateoas.CommunityResource` class (limited to a depth of 1)? I think that in most cases you need the subcommunity information if you fetch the Community information.

Reply:
See also https://jira.duraspace.org/browse/DS-3650

Reply:
We need to find a good way to manage lazy load optimized collections, please check the discussion here: https://wiki.duraspace.org/display/DSPACE/DSpace+7+REST%3A+Coding+DSpace+Objects#DSpace7REST:CodingDSpaceObjects-LinkedRepositoryObject(TODO-proposal)

I suggest to keep the search method subCommunities in place for now as a simple example of searchMethod we can decide later if we want to remove it and provide only a /api/communities/:id/subCommunities endpoint instead (btw, a similar discussion could happen also for the top communities that could be retrieved from the site endpoint)

Reply:
Ok agreed, we can work on this in the lazy loading/projections discussions.